apps/Learning-Buddy/screens/NewQuizScreen.jsx

```diff
@@ -0,0 +1,354 @@
+import { View, Text, ScrollView, StyleSheet } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import React, { useState, useEffect } from 'react';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+import {
+  Card,
+  IconButton,
+  Divider,
+  Paragraph,
+  TextInput
+} from 'react-native-paper';
+import BigButton from '../components/BigButton';
+import { colors } from '../config/colors';
+
+export const NewQuizScreen = () => {
+  const insets = useSafeAreaInsets();
+  const navigation = useNavigation();
+
+  const [text, setText] = useState('');
+  const [numLines, changeNumLines] = useState(1);
+
+  useEffect(() => {
+    const charLen = text.length;
+    const suggestLines = Math.ceil(charLen / 50);
+    suggestLines > 0 ? changeNumLines(suggestLines) : changeNumLines(1);
+  }, [text]);
+
+  const [numQuestions, setNumQuestions] = useState(1);
+
+  const handleIncrement = () => {
+    setNumQuestions(numQuestions + 1);
+  };
+
+  const handleDecrement = () => {
+    if (numQuestions > 1) {
+      setNumQuestions(numQuestions - 1);
+    }
+  };
+
+  const [selectedQuestionType, setSelectedQuestionType] = useState(null);
+
+  const handleQuestionTypePress = (questionTypeContent) => {
+    setSelectedQuestionType(questionTypeContent);
+  };
+
+  const [selectedDifficulty, setSelectedDifficulty] = useState(null);
+
+  const handleDifficultyPress = (difficultyContent) => {
+    setSelectedDifficulty(difficultyContent);
+  };
+
+  return (
+    <View
+      style={{
+        // Paddings to handle safe area
+        paddingTop: insets.top,
+        paddingBottom: insets.bottom,
+        paddingLeft: insets.left,
+        paddingRight: insets.right
+      }}
+    >
+      <ScrollView>
+        <View style={{ flex: 1, marginHorizontal: 35 }}>
+          <View>
+            <Text style={localStyles.pageTitle}>Create New Quiz</Text>
+          </View>
+          <Text style={localStyles.title}>Source Material</Text>
```

Comment:
This text field is behaving strangely; it is cutting off the inputted text and scrolling reveals it. Might have something to do with old numberOfLines/ numLines on 78; take a look and make sure it doesn't cut off things going forward. Experiment with multiple lengths of text, from 100 characters to 200, 300, etc, and make sure it functions well with each.
![Screenshot_1688873444](https://github.com/new-idea-machine/os-lesson-delivery-system/assets/105380332/56a7739b-2963-4a01-a04c-2934f3c54560)
![Screenshot_1688873449](https://github.com/new-idea-machine/os-lesson-delivery-system/assets/105380332/08e8f0cc-abaf-40a3-b8b7-c039592b708d)